react/Viewer/Panel/Expiration.jsx

```diff
@@ -0,0 +1,68 @@
+import React, { useState } from 'react'
+import PropTypes from 'prop-types'
+
+import { useClient, models } from 'cozy-client'
+
+import { formatLocallyDistanceToNow } from '../../I18n/format'
+import Alert from '../../Alert'
+import Button from '../../Buttons'
+import { withViewerLocales } from '../hoc/withViewerLocales'
+import { useI18n } from '../../I18n'
+import ExpirationLink from './ExpirationLink'
+
+const FILES_DOCTYPE = 'io.cozy.files'
+
+const { computeExpirationDate, isExpiringSoon } = models.file
+
+const Expiration = ({ file }) => {
+  const { t } = useI18n()
+  const client = useClient()
+  const [isBusy, setIsBusy] = useState(false)
+
+  const handleClose = async () => {
+    setIsBusy(true)
+    await client.collection(FILES_DOCTYPE).updateMetadataAttribute(file.id, {
```

Comment:
est-ce qu'on devrait pas plutôt faire un client.save() ici ? cc @paultranvan @Crash--

Reply:
J'ai procédé de la manière que dans [`InformationEdit`](https://github.com/cozy/cozy-libs/blob/master/packages/cozy-mespapiers-lib/src/components/Views/InformationEdit.jsx#L50-L59) ou encore dans [`PageEdit`](https://github.com/cozy/cozy-libs/blob/master/packages/cozy-mespapiers-lib/src/components/Views/PageEdit.jsx#L39-L46).

Reply:
Tu peux pas utiliser un save() pour mettre à jour ces infos là pour le moment. 

Par contre, si l'app n'a pas de temps réel sur le io.cozy.files alors le fichier ne sera pas à jour dans le store de cozy-client.

Reply:
Donc pour résumer : 
- est-ce qu'on devrait : oui
- est-ce qu'on peut : non
;)

Reply:
Dis moi comment tu arrives à accéder à `updateMetadataAttribute` depuis cozy-client.

Reply:
En passant par la droite. :trollface:

je veux dire, pourquoi on peut pas faire client.save en passant donc le file qu'on refait "à la main" plutôt qu'en utilisant une méthode https://github.com/cozy/cozy-client/blob/master/packages/cozy-stack-client/src/FileCollection.js#L982 qui prend l'id du doc en arg. ?

Reply:
Parce que c'est un point d'API spécifique pour mettre à jour ces informations... 

On a déjà quelques if sur le `update` https://github.com/cozy/cozy-client/blob/master/packages/cozy-stack-client/src/FileCollection.js#L495-L518 mais il ne gère pas encore le `updateMetadataAttribute`

Reply:
issue / PR welcome :wink:

Reply:
There is already at least one discussion about that https://github.com/cozy/cozy-client/pull/650

Reply:
I didn't see this PR, so I created an issue to link it: https://github.com/cozy/cozy-client/issues/1277